evaluate/eval.py

```diff
@@ -122,19 +122,43 @@ def compute_total_repairs_grdt(self):
         compute_total_repairs_grdt memoizes the number of repairs for cells
         that are specified in the clean/ground truth data. Otherwise repairs
         are defined the same as compute_total_repairs.
+
+        We also distinguish between repairs on correct cells and repairs on
+        incorrect cells (correct cells are cells where init == ground truth).
         """
-        query = "SELECT count(*) FROM " \
-                "  (SELECT _vid_ " \
-                "   FROM   {} as t1, {} as t2, {} as t3 " \
-                "   WHERE  t1._tid_ = t2._tid_ " \
-                "     AND  t1.attribute = t2.attribute " \
-                "     AND  t1.init_value != t2.rv_value " \
-                "     AND  t1._tid_ = t3._tid_ " \
-                "     AND  t1.attribute = t3._attribute_) AS t".format(AuxTables.cell_domain.name,
-                                                                       AuxTables.inf_values_dom.name,
-                                                                       self.clean_data.name)
+        query = """
+        SELECT
+            (t1.init_value = t3._value_) AS is_correct,
+            count(*)
+        FROM   {} as t1, {} as t2, {} as t3
+        WHERE  t1._tid_ = t2._tid_
+          AND  t1.attribute = t2.attribute
+          AND  t1.init_value != t2.rv_value
+          AND  t1._tid_ = t3._tid_
+          AND  t1.attribute = t3._attribute_
+        GROUP BY is_correct
+          """.format(AuxTables.cell_domain.name,
+                  AuxTables.inf_values_dom.name,
+                  self.clean_data.name)
         res = self.ds.engine.execute_query(query)
-        self.total_repairs_grdt = float(res[0][0])
+
+        # Memoize the number of repairs on correct cells and incorrect cells.
```

Comment:
I don't get where the momoization is

Reply:
When I say memoize I refer to the method storing the # of repairs to one of its fields.

I think in we could probably change all these methods to return the value rather than implicitly computing and storing the values to the fields.

Reply:
it is okay for now, we only call it once anyway.